Name collision with ArduinoIoTCloud library

### Describe the problem

When the ArduinoIoTCloud library is used in the same sketch as this library, compilation fails due to a name collision:

```text
In file included from c:\Users\per\Documents\Arduino\libraries\Arduino_UnifiedStorage\src/Arduino_UnifiedStorage.h:10:0,
                 from C:\Users\per\Documents\Arduino\000-support\github\arduino-libraries\Arduino_UnifiedStorage\52\52.ino:2:
c:\Users\per\Documents\Arduino\libraries\Arduino_UnifiedStorage\src/Types.h:6:17: error: redeclaration of 'READ'
 enum FileMode { READ, WRITE, APPEND };
                 ^~~~
In file included from C:\Users\per\Documents\Arduino\000-support\github\arduino-libraries\Arduino_UnifiedStorage\52\52.ino:1:0:
c:\Users\per\Documents\Arduino\libraries\ArduinoIoTCloud\src/ArduinoIoTCloud.h:53:3: note: previous declaration 'permissionType READ'
   READ      = 0x01,
   ^~~~
In file included from c:\Users\per\Documents\Arduino\libraries\Arduino_UnifiedStorage\src/Arduino_UnifiedStorage.h:10:0,
                 from C:\Users\per\Documents\Arduino\000-support\github\arduino-libraries\Arduino_UnifiedStorage\52\52.ino:2:
c:\Users\per\Documents\Arduino\libraries\Arduino_UnifiedStorage\src/Types.h:6:23: error: redeclaration of 'WRITE'
 enum FileMode { READ, WRITE, APPEND };
                       ^~~~~
In file included from C:\Users\per\Documents\Arduino\000-support\github\arduino-libraries\Arduino_UnifiedStorage\52\52.ino:1:0:
c:\Users\per\Documents\Arduino\libraries\ArduinoIoTCloud\src/ArduinoIoTCloud.h:54:3: note: previous declaration 'permissionType WRITE'
   WRITE     = 0x02,
   ^~~~~
```

### To reproduce

Compile the following minimal demonstration sketch:

```cpp
#include <ArduinoIoTCloud.h>
#include <Arduino_UnifiedStorage.h>
void setup() {}
void loop() {}
```

### Expected behavior

Library does not pollute the namespace with declarations of common names. When it is necessary to expose a declaration, a distinctive name (e.g., `Arduino_UnifiedStorageFileMode::READ`) is used.

### Arduino_UnifiedStorage version

d41bdd04f6fd7bc5291efa15ce353e142cacf83b

### Additional context

#### Additional reports

- https://github.com/arduino-libraries/ArduinoIoTCloud/issues/315

--- a/src/Types.h
+++ b/src/Types.h
@@ -3,7 +3,7 @@
 
 #include "Boards.h"
 
-enum FileMode { READ, WRITE, APPEND };
+enum class FileMode { READ, WRITE, APPEND };
 
 #if defined(USES_RENESAS_CORE)
     #include "BlockDevice.h"

--- a/src/UFile.cpp
+++ b/src/UFile.cpp
@@ -21,13 +21,13 @@ bool UFile::open(const char* filename, FileMode fileMode) {
     path = filename;
     // Set the mode based on the fileMode
     switch (fileMode) {
-        case READ:
+        case FileMode::READ:
             mode = "r+";
             break;
-        case WRITE:
+        case FileMode::WRITE:
             mode = "w+";
             break;
-        case APPEND:
+        case FileMode::APPEND:
             mode = "a+";
             break;
         default:

